### PR TITLE
Added all-tables to database via a preCompile step.

### PR DESCRIPTION
**Pull request by ZoranPavlovic**
Added preCompile flag + feature in order to enable project-level parsing/generation. Database.tables now includes all tables parsed, and Table is now the "current" table.

See keegan's changes doc changes as well for more info.

### CODE REVIEW

File: src/main/java/bbd/jportal2/JPortal2Arguments.java [hunk 1]
```diff
@@ -46,7 +49,7 @@
     private List<String> templatePostProcessors = new ArrayList<>();
 
     @Parameter(names = {"--template-location", "-tl"}, description = "Freemarker template location. Default is <current_working_directory>/jportal2_templates")
-    private List<String> templateLocations = Arrays.asList(Paths.get(System.getProperty("user.dir"), "jportal2_templates").toString());
+    private List<String> templateLocations = Arrays.asList(Paths.get(System.getProperty("user.dir"), "jportal_templates").toString());
```
Comment (dieterrosch) [inline src/main/java/bbd/jportal2/JPortal2Arguments.java]: why the change from jportal2_templates to jportal_templates?

File: src/test/java/bbd/jportal2/test/FreeMarkerTest.java [hunk 2]
```diff
@@ -82,31 +85,43 @@ public void testFreeMarkerMultipleSubDirMultipleFiles() throws Exception {
...
+        }
+
+        (new FreeMarker()).generateTemplate(db, tab, TEMPLATE_DIR, nameOfTest, outputDirectory);
 
         //Test output - dir1/file1.txt
         List<String> lines = Files.readAllLines(Paths.get(outputDirectory.toString(), "dir1/file1.txt"));
-        assertEquals("Database Name: jportal_example_db" ,lines.get(0));
-        assertEquals("Field Name: ID"           ,lines.get(1));
-        assertEquals("Static: 1"                ,lines.get(2));
+        assertEquals("Database Name: jportal_example_db",lines.get(0));
```
Comment (dieterrosch) [inline src/test/java/bbd/jportal2/test/FreeMarkerTest.java]: Side note:
We should find a better way to compare the output to expected. I think we should have a directory with the expected output files, and then do a file-based compare rather than a line-by-line string compare. Not important for now, but something to add.

**Comment by dieterrosch** [comment]
Looks good. 2 minor comments.

Please add documentation to explain how to use the projectCompile option and what it does?

**Comment by dieterrosch** [comment]
@ZoranPavlovic Any update on adding documentation on how this works or an example of how to use it?

**Comment by ZoranPavlovic** [comment]
@dieterrosch - The preCompile flag is more for internal use between the classes as I didn't want to deviate too much from the convention that was already there.

I've also added documentation to mention the different variables available inside the freemarker template.

File: src/main/java/bbd/jportal2/JPortal2Arguments.java (hunk 1 above)
Comment (ZoranPavlovic) [reply src/main/java/bbd/jportal2/JPortal2Arguments.java]: Think it snuck in. I've reverted it.

File: src/test/java/bbd/jportal2/test/FreeMarkerTest.java (hunk 2 above)
Comment (ZoranPavlovic) [reply src/test/java/bbd/jportal2/test/FreeMarkerTest.java:101]: I've created an Issue for this so we can make some enhancements. 

https://github.com/SI-Gen/jportal2/issues/69

**Comment by dieterrosch** [comment]
@ZoranPavlovic - Yeah, what I meant it, in the usage doc, just explain what the --projectCompile flag does, and how to use it.

**Comment by dieterrosch** [comment]
I merged this into the VG-Changes branch, which will get pulled into master
